torch/_refs/__init__.py

```diff
@@ -1822,6 +1824,64 @@ def tensor_split(
         return tuple(splits)
 
 
+def hsplit(
+    a: TensorLikeType, indices_or_sections: DimsType
+) -> Tuple[TensorLikeType, ...]:
+    msg = (
+        "torch.hsplit requires a tensor with at least 1 dimension, but got a tensor with "
+        + str(a.ndim)
+        + " dimensions!"
+    )
+    check(a.ndim >= 1, lambda: msg)
+    dim = 0 if a.ndim == 1 else 1
+    if isinstance(indices_or_sections, int):
+        split_size = indices_or_sections
+        msg = (
+            "torch.hsplit attempted to split along dimension "
+            + str(dim)
+            + ", but the size of the dimension "
+            + str(a.shape[dim])
+            + " is not divisible by the split_size "
+            + str(split_size)
+            + "!"
+        )
+        check((split_size != 0 and a.shape[dim] % split_size == 0), lambda: msg)
+        return tensor_split(a, split_size, dim)
+    if not isinstance(indices_or_sections, (list, tuple)):
+        raise TypeError(
+            "hsplit(): received an invalid combination of arguments. "
+            "Expected indices_or_sections to be of type int, list of ints or tuple of ints "
+            f"but got type {type(indices_or_sections)}"
+        )
```

Comment:
Would be nice if `check` can also take the type of error to raise. This would be changed to
`check(isinstance(indices_or_sections, (list, tuple)), msg, error_type=TypeError)`?

Reply:
cc @ezyang -- we're already seeing variants of `check` pop up for different error types

Reply:
Optional third argument sounds good to me